packages/transformer/src/IModelTransformer.ts

```diff
@@ -640,65 +641,46 @@ export class IModelTransformer extends IModelExportHandler {
       LEFT JOIN bis.ExternalSourceAspect esa ON e.ECInstanceId=esa.Element.Id
       WHERE e.ECInstanceId NOT IN (0x1, 0xe, 0x10) -- special static elements
         AND ((Scope.Id IS NULL AND KIND IS NULL) OR (Scope.Id=:scopeId AND Kind=:kind))
-      ORDER BY FederationGuid
     `;
 
+    const provenanceByAspect = new Map<Id64String, Id64String> ();
```

Comment:
@ViliusRuskys I am going to try rewriting this by splitting into two passes, a pass through all fed guids and then a pass through all external source aspects. Since it's reading and not inserting, and new transformations won't add the fed guids, hopefully it won't be too bad

Reply:
@ViliusRuskys it still passes working this way. Can you review and merge at your leisure? Or I'll merge myself tomorrow if you have no protests.

Reply:
I was aware of the increased memory, but was willing to take that sacrifice for cleaner code since the memory is consumed only during the initialize method scope and can be cleaned up right after by the gc.
I'm fine with your approach as well though.
Merging this.